ucsd-ets/edx-platform
# Add support for vimeo videos with course-searchable captions

Modify edx requirements to include our fork of video_xblock and pycaption (whose requirements have been modified to work with our installation, and tested in devstack).

(Including config change for docker-devstack that is needed to test indexing/searching.)

#### Story Link

[EDX-265](https://its-pro.ucsd.edu/browse/EDX-265) - As a UC San Diego Online content developer, I want to be able to link to private Vimeo videos, load captions from the video, and have them searchable in my class. 

#### PR Description

Adds two libraries to edx dependencies, forks of xblock_video and pycaption, which are forked because those packages specify overly strict version requirements on their dependencies which conflict with our EdX installation.

#### Type of change

Please select the options that are relevant.


#### How to test?

With openedx-special on commit a29b18c or newer and PR#16 merged into openedx-config:

Create a course in studio. From Settings > advanced settings, add "video_xblock" to the advanced module list, and save. Then, from any unit, "Add New Component" > Advanced > Video. 

- [ ] Verify that an empty video block has been added, with the placeholder text '[Here be Video]'. 

Click the block's edit link, enter the url for a private Vimeo video, https://vimeo.com/396547794 , and click Save.

- [ ] Verify that the private video displays. 

Click the block's edit link, and click the 'Advanced' button. Verify that the configured API key is present (ending in ...6cd8), though technically you verified that by the video coming up at all. Next to Default Timed Transcript, click the slider on 'Show Available default transcripts'

- [ ] Verify that one transcript is listed, for English. 

Click 'Fetch' next to that transcript.

- [ ] Verify that the drop-down in the 'Enabled Transcripts' section now says 'English'.

Click Save.

- [ ] Verify that the video control bar now shows a [CC] button; that clicking that button shows the 'English' option; and that clicking English adds the English captions to the video
- [ ] Verify that the video control bar now shows a ["] transcript button; that clicking that button shows the transcript in a panel to the right, and that that panel contains a timed transcript for the video. 

Publish the unit, and go to the live version of the course. From the course home, search for a phrase uniqe to this video's transcript, like 'Climate'

- [ ] Verify the video is among the search results.
